server/handler.go

```diff
@@ -239,6 +190,34 @@ func New1ArgHandler[REQUEST, RESPONSE any, CTX HandlerArgument](f func(ctx conte
 	}
 }
 
+// New1ArgNoContentHandler - creates a Handler from a handler function and server.HandlerConfig without body argument (typically Http GET or DELETE) and argument provided by one of Path parameters, Query parameters or Headers
+// i.e. handler function like func OnRequest(ctx context.Context, args api.AdditionalParameters) (*Response[api.YourResponseType], serr.Error)
+// where api.AdditionalParameters would contain extracted values from:
+//    path: i.e. /api/v1/parent/:ID/child/:childID
+//    query: i.e. /api/v1/parent?foo=BAR
+//    headers
+func New1ArgNoContentHandler[RESPONSE any, CTX HandlerArgument](f func(ctx context.Context, arg1 CTX) (*Response[RESPONSE], serr.Error), config HandlerConfig) *Handler2Extensions[Void, RESPONSE, CTX] {
```

Comment:
It might help to provide a type for this function. For example:
```
type HandlerFunc func(ctx context.Context, params CTX) (*Response[RESPONSE], serr.Error)

func NewHandler[RESPONSE any, CTX HandlerArgument](f HandlerFunc, cfg HandlerConfig) Extensions[...] {

}
```

This helps you test later since you can pass in mock functions of the same type and don't have to repeat this signature in every function call. It also lends to readability

I'm also curious to why we're not using variadic arguments in these and don't really understand the need to use a generic. This seems to be a weird mix of factory pattern but also with generics. This could be dropped down to one function potentially:

```
func NewHandler[REQUEST, RESPONSE any](f HandlerFunc, args ...HandlerArgument) {
    // since REQUEST and RESPONSE are 'any' you could use an empty interface here if you don't have a REQUEST
}
```

Again, I am maybe naive since I don't understand the reason to use generics here. Usually, you would use them to support a specific type or interface (usually interface). So passing a type of 'any' doesn't really do anything that you could do by passing this in as 'any' everywhere. It's not really saying anything useful and you could just change it to:
```
type HandlerFunc func(ctx context.Context, args ...HandlerArgument) (*Response[any], serr.Error)

func NewHandler(req, resp any, f HandlerFunc, args ...HandlerArgument) h* handler {

}
```

One more thing to consider, using `any` which is an alias for `interface{}` may not be the best approach here. Usually, you would want to support some kind of named interface or type.

Reply:
this is a bit different - the REQIEST and RESPONSE are generics of type any, as they indeed can be any, as long as they can be unmarshalled and marshalled into JSON, YAML or XML (not sure about the XML support, I'd need to check in source code) or be a stream. The type information is inferred by the compiler when you register specific handler - i.e. 

``
type (
   Input struct {
       ID string 'json: id'
       Payload string 'json: payload'
   }
   Result struct {
      Data string 'json: data'   
  }
)

func HandleFormData(ctx context.Context, body Input) (*Response[Result], serr.Error) { 
  .... do stuff here
}

NewHandler( HandleFormData, HandlerConfig { ... } )
``

We could use any here, but then - casting would be required, which would reduce the code's readability. The input type is passed down the chain, so the infrastructure code knows how to handle the http body - if it should be deserialized into json or should it be considered a byte stream. As this is input data, provided by the http request, that type information would be otherwise unavailable. 

This logic around the handler function is designed to reduce the boilerplate code within the handelr itself, where you'd be normally dealing with raw HttpRequest and be extracting the data out of it, deserializing it into specific type, validating, etc.

Reply:
When it comes to mutli argument variants of the method - there is also a reason why it isn't variant - the ARG1 through ARG3 must be subclasses of HandlerArgument type - this defines where to look for the data to populate these instances, there are following options available:
- headers 
- path parameters
- query parameters
- security context - aka ArmoryPrincipal
Again - all these are available through http's context, but they would require a boiler plate code to extract whenever the handler is defined - these methods are somehow similar to java's or c# handler parameters annotated with [FromPath] [FromQuery] attributes.